releases_test.go

```diff
@@ -0,0 +1,155 @@
+	var documents []string
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line != "---" {
+			document += line + "\n"
+			continue
+		}
+		if document == "" {
+			continue
+		}
+		documents = append(documents, document)
+		document = ""
+	}
+	return documents, nil
+}
+
+// Given a slice of strings defining Release CRs as YAML, this function
+// simply parses the YAML and returns a slice of Releases.
+func parseReleases(documents []string) ([]v1alpha1.Release, error) {
+	var releases []v1alpha1.Release
+	for _, document := range documents {
+		var release v1alpha1.Release
+		err := yaml.Unmarshal([]byte(document), &release)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		releases = append(releases, release)
+	}
+	return releases, nil
+}
+
+// To reuse versionbundle.ValidateIndexReleases, the slice of Releases must first be
+// converted into a slice of versionbundle.IndexRelease.
+func releasesToIndex(releases []v1alpha1.Release) []versionbundle.IndexRelease {
+	var indexReleases []versionbundle.IndexRelease
+	for _, release := range releases {
+		var apps []versionbundle.App
+		for _, app := range release.Spec.Apps {
+			indexApp := versionbundle.App{
+				App:              app.Name,
+				ComponentVersion: app.ComponentVersion,
+				Version:          app.Version,
+			}
+			apps = append(apps, indexApp)
+		}
+		var authorities []versionbundle.Authority
+		for _, component := range release.Spec.Components {
+			indexAuthority := versionbundle.Authority{
+				Name:    component.Name,
+				Version: component.Version,
+			}
+			authorities = append(authorities, indexAuthority)
+		}
+		indexRelease := versionbundle.IndexRelease{
+			Active:      release.Spec.State == "active",
+			Apps:        apps,
+			Authorities: authorities,
+			Date:        release.Spec.Date.Time,
+			Version:     release.Name,
+		}
+		indexReleases = append(indexReleases, indexRelease)
+	}
+	return indexReleases
+}
+
+func Test_Releases(t *testing.T) {
+	testCases := []struct {
+		filename string
+		name     string
+	}{
+		{
+			filename: "aws.yaml",
+			name:     "case 1: aws.yaml is valid",
+		},
+		{
+			filename: "azure.yaml",
+			name:     "case 2: azure.yaml is valid",
+		},
+		{
+			filename: "kvm.yaml",
+			name:     "case 3: kvm.yaml is valid",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			documents, err := scanDocuments(tc.filename)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			releases, err := parseReleases(documents)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			crd := v1alpha1.NewReleaseCRD()
+			var v apiextensions.CustomResourceValidation
+			// Convert the CRD validation into the version-independent form.
+			err = v1beta1.Convert_v1beta1_CustomResourceValidation_To_apiextensions_CustomResourceValidation(crd.Spec.Validation, &v, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			validator, _, err := validation.NewSchemaValidator(&v)
+			if err != nil {
+				t.Fatal(err)
```

Comment:
I would like to do something like this. 

```suggestion
				t.Fatal(microerror.JSON(err))
```

Reply:
Done!